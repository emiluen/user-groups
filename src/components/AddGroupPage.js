import React from 'react';
import { connect } from 'react-redux';
import GroupForm from './GroupForm';
import { startAddGroup } from '../actions/groups';

export class AddGroupPage extends React.Component {
  onSubmit = (group) => {
    this.props.startAddGroup(group);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Group</h1>
          </div>
        </div>
        <div className="content-container">
          <GroupForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddGroup: (group) => dispatch(startAddGroup(group))
});

export default connect(undefined, mapDispatchToProps)(AddGroupPage);
