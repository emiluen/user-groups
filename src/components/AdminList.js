import React from 'react';
import { connect } from 'react-redux';
import List from './List';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
  <List expenses={props.expenses} edit={true} />
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
