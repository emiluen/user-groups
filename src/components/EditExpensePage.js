import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import { startEditGroup } from '../actions/user';

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };
  onSubmit2 = (expense) => {
    this.props.startEditGroup(this.props.expense.id, true);
    this.props.history.push('/');
  };
  onRemove2 = () => {
    this.props.startEditGroup(this.props.expense.id, false);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <button className="button button--secondary" onClick={this.onRemove}>Remove Expense</button>
        </div>
        <div>
          <button className="button" onClick={this.onSubmit2}>Add as My Expense</button>
          <button className="button button--secondary" onClick={this.onRemove2}>Remove as My Expense</button>
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (data) => dispatch(startRemoveExpense(data)),
  startEditGroup: (id, bool) => dispatch(startEditGroup(id, bool))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
