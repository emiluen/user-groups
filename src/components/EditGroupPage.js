import React from 'react';
import { connect } from 'react-redux';
import GroupForm from './GroupForm';
import { startEditGroup, startRemoveGroup } from '../actions/groups';

export class EditGroupPage extends React.Component {
  onSubmit = (group) => {
    this.props.startEditGroup(this.props.group.id, group);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.startRemoveGroup({ id: this.props.group.id });
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Group</h1>
          </div>
        </div>
        <div className="content-container">
          <GroupForm
            group={this.props.group}
            onSubmit={this.onSubmit}
          />
          <button className="button button--secondary" onClick={this.onRemove}>Remove Group</button>
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  group: state.groups.find((group) => group.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  startEditGroup: (id, group) => dispatch(startEditGroup(id, group)),
  startRemoveGroup: (data) => dispatch(startRemoveGroup(data)),
});

export default connect(mapStateToProps, mapDispatchToProps)(EditGroupPage);
