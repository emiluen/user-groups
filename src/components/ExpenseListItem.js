import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({ id, description, amount, createdAt, edit }) => {
  const getLink = () => {
    if (edit) {
      return `/edit/${id}`;
    }

    return `/view/${id}`;
  }

  return (
    <Link className="list-item" to={getLink()}>
      <div>
        <div>Edit: {edit}</div>
        <h3 className="list-item__title">{description}</h3>
        <span className="list-item__sub-title">{moment(createdAt).format('MMMM Do, YYYY')}</span>
      </div>
      <h3 className="list-item__data">{numeral(amount / 100).format('$0,0.00')}</h3>
    </Link>
  );
};

export default ExpenseListItem;
