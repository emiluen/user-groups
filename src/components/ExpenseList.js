import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';
import ExpenseListItem from './ExpenseListItem';

const ExpenseList = (props) => (
  <div className="content-container">
    <div className="list-header">
      <div className="show-for-mobile">Expenses</div>
      <div className="show-for-desktop">Expense</div>
      <div className="show-for-desktop">Amount</div>
    </div>
    <div className="list-body">
      {
        props.expenses.length === 0 ? (
          <div className="list-item list-item--message">
            <span>No expenses</span>
          </div>
        ) : (
            props.expenses.map((expense) => {
              return <ExpenseListItem key={expense.id} {...expense} edit={props.edit} />;
            })
          )
      }
    </div>
  </div>
);

export default ExpenseList;
