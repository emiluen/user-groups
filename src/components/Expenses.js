import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';
import ExpenseList from './ExpenseList';
import ExpenseListItem from './ExpenseListItem';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const Expenses = (props) => (
  <div>
    <ExpensesSummary expenses={props.expenses} edit={props.edit} />
    <ExpenseListFilters />
    <ExpenseList expenses={props.expenses} edit={props.edit} />
  </div>
);

export default Expenses;
