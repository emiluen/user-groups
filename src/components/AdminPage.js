import React from 'react';
import { connect } from 'react-redux';
import Expenses from './Expenses';
import selectExpenses from '../selectors/expenses';

export const AdminPage = (props) => (
  <div>
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">Admin</h1>
      </div>
    </div>
    <div className="content-container">
      <Expenses expenses={props.expenses} edit={true} />
    </div>
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(AdminPage);
