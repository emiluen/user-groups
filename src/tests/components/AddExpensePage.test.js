import React from 'react';
import { shallow } from 'enzyme';
import { AddGroupPage } from '../../components/AddGroupPage';
import expenses from '../fixtures/expenses';

let startAddExpense, history, wrapper;

beforeEach(() => {
  startAddExpense = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(<AddGroupPage startAddExpense={startAddExpense} history={history} />);
});

test('should render AddGroupPage correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit', () => {
  wrapper.find('GroupForm').prop('onSubmit')(expenses[1]);
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(startAddExpense).toHaveBeenLastCalledWith(expenses[1]);
});
