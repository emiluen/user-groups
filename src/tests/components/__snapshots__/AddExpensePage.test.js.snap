// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddGroupPage correctly 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Add Expense
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <GroupForm
      onSubmit={[Function]}
    />
  </div>
</div>
`;
